src/Features/CSharp/Portable/ReplaceDefaultLiteral/CSharpReplaceDefaultLiteralCodeFixProvider.cs

```diff
@@ -0,0 +1,150 @@
+        {
+            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var defaultToken = syntaxRoot.FindToken(span.Start);
+            var defaultLiteral = (LiteralExpressionSyntax)defaultToken.Parent;
+
+            var newRoot = syntaxRoot.ReplaceNode(defaultLiteral, newExpression.WithTriviaFrom(defaultLiteral));
+            return document.WithSyntaxRoot(newRoot);
+        }
+
+        private static (SyntaxNode newExpression, string displayText) GetReplacementExpressionAndText(
+            Document document,
+            LiteralExpressionSyntax defaultLiteral,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            var generator = SyntaxGenerator.GetGenerator(document);
+
+            var type = semanticModel.GetTypeInfo(defaultLiteral, cancellationToken).ConvertedType;
+            if (type != null && type.TypeKind != TypeKind.Error)
+            {
+                if (IsFlagsEnum(type, semanticModel.Compilation) &&
+                    type.GetMembers("None").FirstOrDefault() is IFieldSymbol field && IsZero(field.ConstantValue))
+                {
+                    return GenerateMemberAccess("None");
+                }
+                else if (type.Equals(semanticModel.Compilation.GetTypeByMetadataName("System.Threading.CancellationToken")))
+                {
+                    return GenerateMemberAccess(nameof(CancellationToken.None));
+                }
+                else if (type.SpecialType == SpecialType.System_IntPtr || type.SpecialType == SpecialType.System_UIntPtr)
+                {
+                    return GenerateMemberAccess(nameof(IntPtr.Zero));
+                }
+                else if (semanticModel.GetConstantValue(defaultLiteral, cancellationToken) is var constant && constant.HasValue)
+                {
+                    var newLiteral = generator.LiteralExpression(constant.Value);
+                    return (newLiteral, newLiteral.ToString());
+                }
+                else if (!type.ContainsAnonymousType())
+                {
+                    var defaultExpression = generator.DefaultExpression(type);
+                    return (defaultExpression, $"default({type.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat)})");
+                }
+            }
+
+            return default;
+
+            (SyntaxNode newExpression, string displayText) GenerateMemberAccess(string memberName)
+            {
+                var memberAccess = generator.MemberAccessExpression(generator.TypeExpression(type), memberName);
+                return (memberAccess, $"{type.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat)}.{memberName}");
+            }
+        }
+
+        private static bool IsFlagsEnum(ITypeSymbol type, Compilation compilation)
+        {
+            var flagsAttribute = compilation.GetTypeByMetadataName("System.FlagsAttribute");
```

Comment:
`typeof(FlagsAttribute).FullName`?

Reply:
➡️ Fixed in 1f3cdd5eced9720fa63f287709cf0547ee3f3738